[Bug] [CPU Backend]: vLLM build on Arm CPU fails with pytorch nightly

### Your current environment

<details>
<summary>The output of <code>python collect_env.py</code></summary>

```text
Your output of `python collect_env.py` here
```

</details>


### 🐛 Describe the bug

I tried to build vLLM with PyTorch nightly - version: `2.10.0.dev20251210+cpu` on Arm CPU.
I got this error:
```
CMake Error at cmake/cpu_extension.cmake:225 (find_library):
  Could not find OPEN_MP using the following names: gomp
Call Stack (most recent call first):
  CMakeLists.txt:111 (include)
```

It's caused by a recent change in PyTorch where they stopped shipping versioned/hash-suffixed libgomp (e.g. `libgomp-1dd373e6.so.1.0.0`) and now just ship `libgomp.so.1`. They also now ship it under `torch/libs` instead of `torch.libs`

Our [cmake util func ](https://github.com/vllm-project/vllm/blob/main/cmake/utils.cmake#L148) which searches for PyTorch's libgomp needs to be updated to support these patterns:

- `./lib/python3.10/site-packages/torch.libs/libgomp-a49a47f9.so.1.0.0` - the case with PyTorch 2.9 
- `./lib/python3.10/site-packages/torch/lib/libgomp.so.1` - the case with PyTorch nightly (`2.10.0.dev20251210+cpu`)

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

--- a/cmake/utils.cmake
+++ b/cmake/utils.cmake
@@ -140,16 +140,21 @@ function(vllm_prepare_torch_gomp_shim TORCH_GOMP_SHIM_DIR)
   run_python(_VLLM_TORCH_GOMP_PATH
     "
 import os, glob
-try:
-  import torch
-  torch_pkg = os.path.dirname(torch.__file__)
-  site_root = os.path.dirname(torch_pkg)
-  torch_libs = os.path.join(site_root, 'torch.libs')
-  print(glob.glob(os.path.join(torch_libs, 'libgomp-*.so*'))[0])
-except:
-  print('')
+import torch
+torch_pkg = os.path.dirname(torch.__file__)
+site_root = os.path.dirname(torch_pkg)
+
+# Search both torch.libs and torch/lib
+roots = [os.path.join(site_root, 'torch.libs'), os.path.join(torch_pkg, 'lib')]
+candidates = []
+for root in roots:
+    if not os.path.isdir(root):
+        continue
+    candidates.extend(glob.glob(os.path.join(root, 'libgomp*.so*')))
+
+print(candidates[0] if candidates else '')
 "
-    "failed to probe torch.libs for libgomp")
+    "failed to probe for libgomp")
 
   if(_VLLM_TORCH_GOMP_PATH STREQUAL "" OR NOT EXISTS "${_VLLM_TORCH_GOMP_PATH}")
     return()

--- a/vllm/platforms/cpu.py
+++ b/vllm/platforms/cpu.py
@@ -325,10 +325,16 @@ class CpuPlatform(Platform):
             # We need to find the location of PyTorch's libgomp
             torch_pkg = os.path.dirname(torch.__file__)
             site_root = os.path.dirname(torch_pkg)
-            torch_libs = os.path.join(site_root, "torch.libs")
-            pytorch_libgomp_so_candidates = glob.glob(
-                os.path.join(torch_libs, "libgomp-*.so*")
-            )
+            # Search both torch.libs and torch/lib - See: https://github.com/vllm-project/vllm/issues/30470
+            torch_libs_paths = [
+                os.path.join(site_root, "torch.libs"),
+                os.path.join(torch_pkg, "lib"),
+            ]
+            pytorch_libgomp_so_candidates = []
+            for torch_libs in torch_libs_paths:
+                pytorch_libgomp_so_candidates.extend(
+                    glob.glob(os.path.join(torch_libs, "libgomp*.so*"))
+                )
             if pytorch_libgomp_so_candidates:
                 pytorch_libgomp_so = pytorch_libgomp_so_candidates[0]
                 if ld_preload_str:

